File: 2020/01/main.py
import itertools
import math

with open('input.txt', 'r') as input_file:
    input_numbers = [int(line.strip()) for line in input_file.readlines()]


def solve(numbers, num_of_entries):
    combis = itertools.combinations(numbers, num_of_entries)
    values = next(combis)
    while sum(values) != 2020:
        values = next(combis)
    return math.prod(values)


print(solve(input_numbers, 2))
print(solve(input_numbers, 3))

File: 2020/01/rust/src/main.rs
use std::fs::File;
use std::io::{BufRead, BufReader};
use itertools::Itertools;

fn main() {
    let input_file = File::open(r"..\input.txt").unwrap();
    let numbers: Vec<u32> = BufReader::new(input_file)
        .lines()
        .map(|l| l.unwrap().parse::<u32>().unwrap())
        .collect();

    let solution1 = solve(&numbers, 2);
    println!("{}", solution1);

    let solution2 = solve(&numbers, 3);
    println!("{}", solution2);
}

fn solve(numbers: &Vec<u32>, num_of_entries: usize) -> u32{
    numbers
        .iter()
        .combinations(num_of_entries)
        .find(|c| c.iter().map(|x| *x).sum::<u32>() == 2020)
        .unwrap()
        .iter()
        .fold(1, |a, b| a * *b)
}